File: src/main/java/easy/BestTimeToBuy.java
package easy;

public class BestTimeToBuy {

    public int maxProfit(int[] prices) {
        if (prices == null || prices.length < 2) {
            return 0;
        }

        int minPrice = Integer.MAX_VALUE;
        int maxProfit = 0;

        for (int price : prices) {
            if (price < minPrice) {
                minPrice = price;
            } else if (price - minPrice > maxProfit) {
                maxProfit = price - minPrice;
            }
        }

        return maxProfit;
    }

    public static void main(String[] args) {
        BestTimeToBuy buy = new BestTimeToBuy();
        int[] prices = {1, 3, 1, 7, 6};
        int profit = buy.maxProfit(prices);
        System.out.println(profit);
    }

}

File: src/main/java/easy/kotlin/BestTimeToBuy.kt
package easy.kotlin

class BestTimeToBuy {

    fun maxProfit(prices: IntArray) : Int {
        if (prices.size < 2) {
            return 0
        }

        var minPrice = Int.MAX_VALUE
        var maxProfit = 0

        for (price in prices) {
            if (price < minPrice) {
                minPrice = price
            } else if (price - minPrice > maxProfit) {
                maxProfit = price - minPrice
            }
        }

        return maxProfit
    }
}